### PR TITLE
stream webdav downloads using http client

### PR DESCRIPTION
**Pull request by icewind1991**
Extends the http client to allow streamed responses and uses that for dav downloads.

Replaces #10620

cc @PVince81 @LukasReschke @MorrisJobke

### CODE REVIEW

**Comment by Xenopathic** [comment]
Why use Guzzle instead of the Sabre DAV Client that is already used in that class? It supports streams as well, with nearly the same interface:

```
$request = new \Sabre\HTTP\Request('GET', $this->createBaseUri() . $this->encodePath($path));

$response = $this->client->send($request);

return $response->getBodyAsStream();
```

Also, you missed the cURL stuff in `uploadFile()`.

**Comment by icewind1991** [comment]
From what I can tell that api doesn't actually stream the result, just returns the string result as a stream

cc @evert

**Comment by evert** [comment]
This is true.. It's been a feature request for a while: https://github.com/fruux/sabre-http/issues/15

**Comment by Xenopathic** [comment]
@evert As @DeepDiver1975 sometimes says... FIXIT! :laughing:

**Comment by evert** [comment]
I don't know how!

**Comment by icewind1991** [comment]
@Xenopathic @PVince81 @DeepDiver1975 can this be merged?

File: lib/private/http/client/response.php [hunk 1]
```diff
@@ -34,17 +34,26 @@ class Response implements IResponse {
 	private $response;
 
 	/**
+	 * @var bool
+	 */
+	private $stream;
+
+	/**
 	 * @param GuzzleResponse $response
+	 * @param bool $stream
```
Comment (PVince81) [inline lib/private/http/client/response.php]: `$isStream` ? Is it a boolean or can it be an actual stream ?

**Comment by PVince81** [comment]
Seems to work with federated shares :+1: 

Would be good to find out whether it works with valid SSL certs though. @LukasReschke what do you think ?

File: lib/private/http/client/response.php (hunk 1 above)
Comment (icewind1991) [inline lib/private/http/client/response.php]: A boolean, to tell the Response class if it should treat the response as a stream

**Comment by Xenopathic** [comment]
Still missing the other cURL stuff in `uploadFile()`

**Comment by Xenopathic** [comment]
:+1:

**Comment by Xenopathic** [comment]
:+1:

**Comment by icewind1991** [comment]
@PVince81 can you re-review with the latest changes

**Comment by icewind1991** [comment]
@DeepDiver1975 @MorrisJobke can you review this?

**Comment by LukasReschke** [comment]
:+1:

**Pull request merged**